### PR TITLE
JAVA-3116: update surefire/failsafe to 3.0.0 to fix issue running tests with specified jvm

### PR DESCRIPTION
**Pull request by hhughes**
Additionally:
- Set --jvm_version=8 when running dse 6.8.19+ with graph workloads (DSP-23501)
- Update commons-configuration2 to 2.9.0 + deps in BundleOptions to support java17
- Update felix framework version to 7.0.1 for java17 (FELIX-6287)
- Pick up newer bndlib for ArrayIndexOutOfBounds error printed with OsgiGraphIT (bndtools/bnd issue#3405)
  - Update pax-url-wrap to 2.6.4 (and bring pax-url-reference up to the same version)
  - Force newer tinybundles version 3.0.0 (default 2.1.1 version required older bndlib)

### CODE REVIEW

**Comment by hhughes** [comment]
Test failures appear to caused by flaky `ccm start` in `DefaultLoadBalancingPolicyIT` + [JAVA-3117](https://datastax-oss.atlassian.net/browse/JAVA-3117)

[JAVA-3117]: https://datastax-oss.atlassian.net/browse/JAVA-3117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: pom.xml [hunk 1]
```diff
@@ -269,12 +270,17 @@
...
       <dependency>
         <groupId>org.ops4j.pax.url</groupId>
         <artifactId>pax-url-reference</artifactId>
-        <version>2.6.2</version>
+        <version>${pax-url.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.ops4j.pax.tinybundles</groupId>
+        <artifactId>tinybundles</artifactId>
+        <version>3.0.0</version>
```
Comment (absurdfarce) [inline pom.xml:283]: Just to make sure my understanding is correct: all the version upgrades (both in this pom as well as in BundleOptions.java) are focused only on upgrading deps for the standalone OSGi tests, right?  We're not updating anything here that would be visible to users of the Java driver (whether they use OSGi or not), correct?

File: test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java [hunk 2]
```diff
@@ -296,7 +300,10 @@ public void resume(int n) {
   }
 
   public void start(int n) {
-    execute("node" + n, "start");
+    List<String> cmdAndArgs = Lists.newArrayList("node" + n, "start");
+    overrideJvmVersionForDseWorkloads()
+        .ifPresent(jvmVersion -> cmdAndArgs.add(String.format("--jvm_version=%d", jvmVersion)));
+    execute(cmdAndArgs.toArray(new String[0]));
```
Comment (absurdfarce) [inline test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java:306]: I don't _love_ the duplication of the --jvm_version handling here across two instances of start() but fixing it is harder than one might like.  There's a dependency not only on class member fields (in the form of jvmArgs) but also method call args (in the form of  n in the node case) that make abstracting it kinda ugly.  It's prolly okay to go with for now; we can always improve it later if somebody hits upon a good idea for doing so.

**Review by absurdfarce — APPROVED** [review]
A few small notes, neither of them nearly enough to request changes here.  Unless my understanding of the goal here is considerably different than what it actually is this should work quite well.

File: pom.xml (hunk 1 above)
Comment (hhughes) [reply pom.xml:283]:
```
% mvn dependency:tree -Dincludes=org.ops4j.pax.tinybundles:tinybundles
[INFO] --- dependency:3.1.2:tree (default-cli) @ java-driver-osgi-tests ---
[INFO] com.datastax.oss:java-driver-osgi-tests:jar:4.17.1-SNAPSHOT
[INFO] \- org.ops4j.pax.exam:pax-exam-junit4:jar:4.13.4:test
[INFO]    \- org.ops4j.pax.exam:pax-exam-spi:jar:4.13.4:test
[INFO]       \- org.ops4j.pax.tinybundles:tinybundles:jar:3.0.0:test
[INFO] 
```

`tinybundles` is only used by osgi-tests under the `test` scope

File: test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java (hunk 2 above)
Comment (hhughes) [reply test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/CcmBridge.java:306]: Agreed. This is just test code so we can tidy it up later if needed.

**Pull request merged**